front-end/src/screens/Settings/username.tsx

```diff
@@ -70,30 +78,61 @@ const Username = ({ className }:Props): JSX.Element => {
 	};
 
 	return (
-		<Form standalone={false} className={className}>
+		<Form standalone={false} className={className} onSubmit={handleSubmit(handleSubmitForm)}>
 			<Form.Group>
 				<Form.Field width={10}>
 					<label>Username</label>
+					{editing ?
+						<input
+							className={errors.username ? 'error' : ''}
+							defaultValue={username || ''}
+							name='username'
+							placeholder='username'
+							ref={register({ maxLength:30, minLength:3, pattern: /^[A-Za-z0-9._-]*$/, required: true })}
```

Comment:
It'd be better to share those values rather than duplicate them.

Reply:
done